example-generator/src/main/java/org/ehrbase/openehr/sdk/examplegenerator/ExampleGeneratorConfig.java

```diff
@@ -200,15 +204,19 @@ private static int determineIndex(double t, int size) {
         return Math.min(Math.max(0, (int) (t * size)), size - 1);
     }
 
-    private static Optional<WebTemplateInput> getInput(WebTemplateNode node, String suffix) {
-        return Optional.ofNullable(node)
-                .map(WebTemplateNode::getInputs)
-                .stream()
-                .flatMap(List::stream)
-                .filter(e -> Objects.equals(suffix, e.getSuffix()))
-                .findFirst();
+    private static Optional<WebTemplateInput> getInput(WebTemplateNode node, Predicate<WebTemplateInput> predicate) {
```

Comment:
Why dillute the signature?
The predicate seems to be always the same.

Reply:
The getInputs is the more generic method now